include/exadg/grid/calculate_characteristic_element_length.h

```diff
@@ -55,16 +59,55 @@ calculate_minimum_vertex_distance(dealii::Triangulation<dim> const & triangulati
   return dealii::Utilities::MPI::min(min_cell_diameter, mpi_comm);
 }
 
+/**
+ * This function calculates a characteristic element length given the volume of the element, the
+ * number of space dimensions, and the element type.
+ */
+template<typename Number>
+inline Number
+calculate_characteristic_element_length(Number const        element_volume,
+                                        unsigned int const  dim,
+                                        ElementType const & element_type)
+{
+  if(element_type == ElementType::Hypercube)
+  {
+    // calculate the length h of a dim-dimensional cube that has the volume "element_volume".
+    return std::exp(std::log(element_volume) / (double)dim);
+  }
+  else if(element_type == ElementType::Simplex)
+  {
+    // calculate the length h of an equilateral triangle/tetrahedron that has the volume
+    // "element_volume" (V = h^dim / factor).
+    double factor = 1.0;
+
+    if(dim == 2)
+      factor = 4.0 / std::sqrt(3.0);
+    else if(dim == 3)
+      factor = 6.0 * std::sqrt(2.0);
+    else
+      AssertThrow(false, dealii::ExcMessage("Only dim = 2, 3 implemented."));
+
+    return std::exp(std::log(element_volume * factor) / (double)dim);
+  }
+  else
+  {
+    AssertThrow(false,
+                dealii::ExcMessage("This function is not implemented for the given ElementType."));
+    return std::exp(std::log(element_volume) / (double)dim);
+  }
+}
+
 /**
  * This function calculates a characteristic resolution limit for high-order Lagrange polynomials
  * given a mesh size h. This one-dimensional resolution limit is calculated as the grid size divided
  * by the number of nodes per coordinate direction. Hence, the result depends on the function space
  * (H^1 vs. L^2).
  */
-inline double
-calculate_characteristic_element_length(double const       element_length,
-                                        unsigned int const fe_degree,
-                                        bool const         is_dg)
+template<typename Number>
+inline Number
+calculate_high_order_element_length(Number const       element_length,
+                                    unsigned int const fe_degree,
+                                    bool const         is_dg)
 {
   unsigned int n_nodes_1d = fe_degree;
 
```

Comment:
interesting dependency on `is_dg` ... the number of node-to-node segments is always fe_degree+1, or what is the point here??

Reply:
What we calculate here is the distance between nodes, assuming an equidistant distribution of nodes.

Reply:
> interesting dependency on `is_dg` ... the number of node-to-node segments is always fe_degree+1, or what is the point here??

i meant fe_degree-1, but nevermind, we talked about this in person ;)